hooks/tk-multi-publish2/basic/publish_session.py

```diff
@@ -0,0 +1,712 @@
+        :returns: True if item is valid, False otherwise.
+        """
+
+        publisher = self.parent
+        path = _session_path()
+
+        if not path:
+            # the session still requires saving. provide a save button.
+            # validation fails.
+            self.logger.error(
+                "The Maya session has not been saved.",
+                extra=_get_save_as_action()
+            )
+            return False
+
+        # ensure we have an updated project root
+        project_root = cmds.workspace(q=True, rootDirectory=True)
+        item.properties["project_root"] = project_root
+
+        # warn if no project root could be determined.
+        if not project_root:
+            self.logger.warning(
+                "Your session is not part of a maya project.",
+                extra={
+                    "action_button": {
+                        "label": "Set Project",
+                        "tooltip": "Set the maya project",
+                        "callback": lambda: mel.eval('setProject ""')
+                    }
+                }
+            )
+
+        # get the path in a normalized state. no trailing separator,
+        # separators are appropriate for current os, no double separators,
+        # etc.
+        path = sgtk.util.ShotgunPath.normalize(path)
+
+        # determine the publish path, version, type, and name
+        publish_info = self._get_publish_info(path, settings)
+
+        publish_name = publish_info["publish_name"]
+
+        # see if there are any other publishes of this path with a status.
+        # Note the name, context, and path *must* match the values supplied to
+        # register_publish in the publish phase in order for this to return an
+        # accurate list of previous publishes of this file.
+        publishes = publisher.util.get_conflicting_publishes(
+            item.context,
+            path,
+            publish_name,
+            filters=["sg_status_list", "is_not", None]
+        )
+
+        if publishes:
+            conflict_info = (
+                "If you continue, these conflicting publishes will no longer "
+                "be available to other users via the loader:<br>"
+                "<pre>%s</pre>" % (pprint.pformat(publishes),)
+            )
+            self.logger.warn(
+                "Found %s conflicting publishes in Shotgun" %
+                (len(publishes),),
+                extra={
+                    "action_show_more_info": {
+                        "label": "Show Conflicts",
+                        "tooltip": "Show the conflicting publishes in Shotgun",
+                        "text": conflict_info
+                    }
+                }
+            )
+
+        # check to see if the next version of the work file already exists on
+        # disk. if so, warn the user and provide the ability to jump to save
+        # to that version now
+        (next_version_path, version) = self._get_next_version_info(
```

Comment:
Similarly, calculating the next version of the workfile path is moved into this new method which handles both the ZC and template workflows.